examples/scripts/TreeCompareAuto.C

```diff
@@ -0,0 +1,309 @@
+        } else {
+            outstream << "CHecking Kolmogorov" << endl;
+            leafIsIdentical = CheckHistogramKolmogorov(htemp, hist1, outstream);
+        }
+
+        if (leafIsIdentical) {
+            numKolmogorovHistograms++;
+            outstream << "**********************" << std::endl;
+            hist1->Clear();
+            hist->Clear();
+            htemp->Clear();
+            continue;
+        } else {
+            outstream << "Data are differnt" << std::endl;
+            outstream << "**********************" << std::endl;
+            outstream << "Entries: " << hist->GetEntries() << std::endl;
+            outstream << "Mean: " << hist->GetMean() << std::endl;
+            outstream << "RMS: " << hist->GetRMS() << std::endl;
+            outstream << "Underflow: " << hist->GetBinContent(0) << std::endl;
+            outstream << "Overflow: " << hist->GetBinContent(hist->GetNbinsX() + 1) << std::endl;
+            outstream << "**********************" << std::endl;
+            okay = kFALSE;
+            numFailedLeaves++;
+        }
+    }
+
+    if (!okay) {
+        std::cout << outstream.str();
+        std::cout << "Test failed." << std::endl;
+        std::cout << numFailedLeaves << " of " << numTestedLeaves << " leaves are different." << std::endl;
+        return 1;
+    }
+    //  std::cout << outstream.str();
+    std::cout << "Tested leaves:                    " << numTestedLeaves << std::endl;
+    //  std::cout << "Empty leaves:                     " << numEmptyLeaves << std::endl;
+    std::cout << "Leaves with identical entries:    " << numIdenticalEntries << std::endl;
+    std::cout << "Leaves with identical histograms: " << numIdenticalHistograms << std::endl;
+    std::cout << "Leaves with kolmo histograms:     " << numKolmogorovHistograms << std::endl;
+    std::cout << "Test passed. All leaves of all branches are exactly identical." << std::endl;
+
+    return 0;
+}
+
+Bool_t CheckEntriesIdentical(TH1* compHist, std::stringstream& outstream)
+{
+    if ((TMath::Abs(compHist->GetMean()) > 0.000001 && TMath::Abs(compHist->GetRMS()) > 0.000001)
+        || (0 != compHist->GetBinContent(0)) || (0 != compHist->GetBinContent(compHist->GetNbinsX() + 1))) {
+        return kFALSE;
+    } else {
+        outstream << "Entries are identical." << std::endl;
+        outstream << "**********************" << std::endl;
+        return kTRUE;
+    }
+}
+
+Bool_t CheckHistogramIdentical(TH1* origHist, TH1* newHist, std::stringstream& outstream)
+{
+    if (origHist && newHist) {
+        outstream << "Comparing histograms" << std::endl;
+        for (int x = 0; x < origHist->GetNbinsX() + 1; ++x) {
+            if (origHist->GetBinContent(x) != newHist->GetBinContent(x)) {
+                return kFALSE;
+            }
+        }
+        outstream << "Histograms are identical." << std::endl;
+        outstream << "**********************" << std::endl;
+        return kTRUE;
+    }
+    return kFALSE;
+}
```

Comment:
```suggestion
bool operator==(TH1 const& lhs, TH1X const& rhs)
{
    for (int x = 0; x < lhs.GetNbinsX() + 1; ++x) {
        if (lhs.GetBinContent(x) != rhs.GetBinContent(x)) {
            return false;
        }
    }
    return true;
}
```

* Remove `CheckHistogramIdentical()` and check my corresponding comment at the call site

Reply:
Done